amanuensis/resources/admin/admin_associated_user.py

```diff
@@ -25,10 +29,73 @@ def add_associated_users(users):
     with flask.current_app.db.session as session:
         associated_user_schema = AssociatedUserSchema(many=True)
         ret = []
-        for user in users:
-            ret.append(udm.add_associated_user(session, user["project_id"], user["email"] if "email" in user else None, user["id"] if "id" in user else None))
-        associated_user_schema.dump(ret)
-        return ret
+        users_with_email = [user for user in users if "email" in user]
+        users_with_id = [
+            user
+            for user in users
+            if "id" in user
+        ]
+        if not (users_with_email or users_with_id):
+            raise UserError("The associated user must have at least an email or id. Neither was found.")
+        user_emails = [user["email"] for user in users_with_email]
+        user_ids = [user["id"] for user in users_with_id]
+        associated_users = udm.get_associated_users(session, user_emails)
+        associated_users += udm.get_associated_users_by_id(session, user_ids)
```

Comment:
This logic can generate duplicated users. For instance if a user has both email and id it will be added twice.

Reply:
Lines 75-86 check for duplicates and skip those.